app/javascript/react/api/apiEndpoints.ts

```diff
@@ -22,4 +26,30 @@ export const API_ENDPOINTS: ApiEndpoints = {
   fetchSelectedDataRangeOfStream: (id, startDate, endDate) =>
     `/stream_daily_averages?stream_id=${id}&start_date=${startDate}&end_date=${endDate}`,
   fetchThresholds: (filters) => `/thresholds/${filters}`,
+  fetchUsernames: (username) => `/autocomplete/usernames?q[input]=${username}`,
+  fetchTags: (params) => {
+    let url = `/${
+      params.sessionType
+    }/autocomplete/tags?q[input]=${encodeURIComponent(
+      params.tag
+    )}&q[west]=${encodeURIComponent(params.west)}&q[east]=${encodeURIComponent(
+      params.east
+    )}&q[south]=${encodeURIComponent(
+      params.south
+    )}&q[north]=${encodeURIComponent(
+      params.north
+    )}&q[time_from]=${encodeURIComponent(
+      params.timeFrom
+    )}&q[time_to]=${encodeURIComponent(params.timeTo)}&q[usernames]=${
+      params.usernames
+    }&q[sensor_name]=${encodeURIComponent(
+      params.sensorName
+    )}&q[unit_symbol]=${encodeURIComponent(params.unitSymbol)}`;
+
+    if (params.sessionType === "fixed") {
+      url += "&q[is_indoor]=false&q[is_active]=true";
```

Comment:
We need to include a comment indicating that these fixed parameters are temporary and need to be passed from the component.